### PR TITLE
EmulatedController: encapsulate default device behind getters/setters

### PR DESCRIPTION
**Pull request by ligfx**
And remove the requirement to call `UpdateDefaultDevice` after setting the default device. Two steps to set a variable is no fun!

### CODE REVIEW

File: Source/Core/InputCommon/ControllerEmu/ControllerEmu.h [hunk 1]
```diff
@@ -32,7 +32,10 @@ class EmulatedController
 
   virtual void LoadConfig(IniFile::Section* sec, const std::string& base = "");
   virtual void SaveConfig(IniFile::Section* sec, const std::string& base = "");
-  void UpdateDefaultDevice();
+
+  const ciface::Core::DeviceQualifier& GetDefaultDevice();
```
Comment (lioncash) [inline Source/Core/InputCommon/ControllerEmu/ControllerEmu.h]: Shouldn't the member function be const qualified as well?

File: Source/Core/InputCommon/ControllerEmu/ControllerEmu.cpp [hunk 2]
```diff
@@ -48,29 +48,42 @@ void EmulatedController::UpdateReferences(const ControllerInterface& devi)
...
+const ciface::Core::DeviceQualifier& EmulatedController::GetDefaultDevice()
 {
+  return m_default_device;
+}
+
+void EmulatedController::SetDefaultDevice(const std::string& device)
+{
+  ciface::Core::DeviceQualifier devq;
+  devq.FromString(device);
+  SetDefaultDevice(devq);
```
Comment (lioncash) [inline Source/Core/InputCommon/ControllerEmu/ControllerEmu.cpp]: `devq` can be `std::move`'d here as well, no?

File: Source/Core/InputCommon/ControllerEmu/ControllerEmu.h (hunk 1 above)
Comment (ligfx) [reply Source/Core/InputCommon/ControllerEmu/ControllerEmu.h]: Yeah, I missed that.

File: Source/Core/InputCommon/ControllerEmu/ControllerEmu.cpp (hunk 2 above)
Comment (ligfx) [reply Source/Core/InputCommon/ControllerEmu/ControllerEmu.cpp]: Yep. I originally had that, but misattributed a different compile error to it. I've put `std::move` back in.

**Pull request merged**